### DISCUSSION TITLE
Scanning Ubuntu Chiseled Images Now Supported

### DISCUSSION THREAD

**Discussion by mthalman** [dotnet/dotnet-docker#4739] [Announcements]
# Scanning Ubuntu Chiseled Images Now Supported

We've worked with our partners at Canonical to provide a scanning solution for [Ubuntu Chiseled images](https://devblogs.microsoft.com/dotnet/dotnet-6-is-now-in-ubuntu-2204/#net-in-chiseled-ubuntu-containers). Scanning support is the last step before fully supporting Ubuntu Chiseled images in production.

Scanning tools typically examine Linux package databases located in the container file system in order to determine which packages are installed. These tools didn't work for Ubuntu Chiseled images because the package databases were removed as part of the chiseling process.

Our [solution](https://github.com/dotnet/dotnet-docker/pull/4688) generates a dpkg status file that's stored in the image. This file is a common format known by scanning tools and describes the packages (technically, they're [slices](https://github.com/canonical/chisel)) that are installed.

We've published updated Chiseled images that support scanning:

* `mcr.microsoft.com/dotnet/<image-type>:8.0-preview-jammy-chiseled`
* `mcr.microsoft.com/dotnet/nightly/<image-type>:6.0-jammy-chiseled`
* `mcr.microsoft.com/dotnet/nightly/<image-type>:7.0-jammy-chiseled`

Note: These images are intended for testing. We'll publish an announcement when the Ubuntu Chiseled images are supported in production.

We understand that auditing container images is an important pre-requisite to consuming these images. Installed package information is necessary for vulnerability scanning and SBOM production. Please try out these images and scan them with your favorite tool and let us know your results.

**Comment by akoken** [comment]
Hi @mthalman 

It is an essential addition, particularly for us. I have thoroughly tested the Ubuntu Chiseled images in our test environment, and it seems that the SBOM functionality is working flawlessly. The results far surpass those of the regular jammy images, making a significant improvement.

Thank you once again for your efforts.

**Image:** mcr.microsoft.com/dotnet/nightly/aspnet:7.0-jammy-chiseled-amd64
Scanner: Trivy@v0.32.1

![image](https://github.com/dotnet/dotnet-docker/assets/2880254/27fb1206-f29d-4ea1-9fe5-6cf1b0df3e09)

**Comment by gareth-houghton** [comment]
Hi @mthalman 

The outputs look to be really good. Even better to see the runtime-deps image have just the 7 packages in my SBOM output! Great work getting it so small

**Comment by mathieu-benoit** [comment]
Indeed! [Great to see](https://medium.com/google-cloud/sail-sharp-8-tips-to-optimize-and-secure-your-net-containers-for-kubernetes-c68ba253844a?sk=4bd6bac66b6ee5ea0ee4919e70c0f205) that we could now look at the details of the packages with `syft` for example, it was not the case before:
```bash
syft mcr.microsoft.com/dotnet/runtime-deps:8.0-jammy-chiseled
```
```
 ✔ Loaded image                                                                                                                                                                                        mcr.microsoft.com/dotnet/runtime-deps:8.0-jammy-chiseled
 ✔ Parsed image                                                                                                                                                                         sha256:701977f1dd3b8738f025d1fe784271575c13dca7fbc137ea8982edfb74e54286
 ✔ Cataloged packages              [7 packages]  
NAME             VERSION                   TYPE 
base-files       12ubuntu4.4               deb   
ca-certificates  20230311ubuntu0.22.04.1   deb   
libc6            2.35-0ubuntu3.1           deb   
libgcc-s1        12.3.0-1ubuntu1~22.04     deb   
libssl3          3.0.2-0ubuntu1.10         deb   
libstdc++6       12.3.0-1ubuntu1~22.04     deb   
zlib1g           1:1.2.11.dfsg-2ubuntu9.2  deb
```

Thanks for that!
